src/ui/control/geolocate_control.js

```diff
@@ -120,6 +127,7 @@ class GeolocateControl extends Evented {
         this._map = map;
         this._container = DOM.create('div', `mapboxgl-ctrl mapboxgl-ctrl-group`);
         checkGeolocationSupport(this._setupUI);
+        this._map.on('move', this._onMove);
```

Comment:
Why do we need to watch for map move? The real world radius in meters to pixel conversion depends on latitude of the user location, not the latitude of the map.

Reply:
Yup -- had a hard time finding documentation on these events. Updated.